### DISCUSSION TITLE
Is there a way to update branch protection rule's pattern from master to main using gh cli?

### DISCUSSION THREAD

**Discussion by sukumarp2022** [community/community#44582] [Enterprise]
### Github branch protection rules

Question

### Body

I'm trying to migrate a repo which has branch protection rules from Github Enterprise Server to Github Enterprise Cloud.

After cloning and pushing the mirror repo using git clone --mirror and git push --mirror commands.

After this clone and push, I'm trying to rename branches in my local (from master to main).

I already have branch protection rules pushed as part of push --mirror. and If I rename the branch locally from master to main, it's throwing error because the master branch itself gets deleted entirely.

I'm looking if there is a way to edit the the protection rules locally in the mirror repo even before pushing it to Github Cloud.

I did look at the contents of the mirror repo. But can't seem to find where the branch protection rules exists in mirror repo if any.

How do I resolve this?

**Comment by jge162** [comment]
It sounds like you're running into a problem with the branch protection rules when trying to rename the "master" branch in your local mirror repository to "main". Since the branch protection rules are stored on the remote repository, renaming the branch in your local repository won't change the rules on the remote.

One solution to this problem would be to rename the "master" branch to "main" on the remote repository before renaming it locally. You can do this by navigating to the "Branches" page on Github Enterprise Cloud, and selecting the "master" branch. From there, you should be able to click on the "..." button and select "Rename", and enter "main" as the new name.

After renaming the branch on the remote repository, you can then rename the "master" branch in your local repository to "main" using the git command git branch -m master main.

You can also use Github REST API to update the protection rules of a branch, in case you want to make some changes to it.

You can also check if your git version supports renaming branch while preserving the branch protection rules.
